apps/mobile/src/contexts/ManageWalletContext.tsx

```diff
@@ -107,6 +117,11 @@ const ManageWalletProvider = memo(({ children }: Props) => {
       }
 
       bottomSheet.current?.dismiss();
+
+      if (onSuccessRef.current) {
+        onSuccessRef.current();
+        onSuccessRef.current = null;
```

Comment:
Its kind of hacky solution 🤦‍♂️

Basically, we passed the `onSuccess` callback to the `openManageWallet` methods inside the `ManageWalletContext`

```tsx
      openManageWallet({
        title: 'You need to connect a wallet to post',
        onSuccess: () => {
          handleCreatePost();
        },
      });
```

We wanna call the `onSuccess` not on `openManageWallet` but in the `handleSignMessage` which was triggered by `useEffect`


**Reference:**
https://github.com/gallery-so/gallery/blob/jakz%2Fupsell-banner-no-wallet/apps/mobile/src/contexts/ManageWalletContext.tsx#L82:L139


Open to any solution/advice 🙏

The reason why I'm using `ref` instead of `useState` because we don't need to show the data in the UI, so it can saves the re-render stuff

Reply:
ohhh ok, gotcha! I don't think it's that hacky then, I think if it used useState that would be a pretty standard way of doing this. and using `ref` to avoid re-renders since we don't need it for display makes sense to me. it might be worth adding a short comment to explain 

will also give robin a chance to see this and see if he agrees!